datafusion/physical-plan/src/joins/nested_loop_join.rs

```diff
@@ -75,21 +75,22 @@ struct JoinLeftData {
     probe_threads_counter: AtomicUsize,
     /// Memory reservation for tracking batch and bitmap
     /// Cleared on `JoinLeftData` drop
-    _reservation: MemoryReservation,
+    #[allow(dead_code)]
+    reservation: MemoryReservation,
```

Comment:
```suggestion
    /// reservation is cleared on Drop 
    #[expect(dead_code)]    
    reservation: MemoryReservation,
```

Also if you use `expect(` rather than `allow` clippy will error if the code ever changes (and we can remove the unecessary annotation). Allow will still run

Reply:
ooh, this is nice